File: scala/src/test/scala/com/laamella/code_state_machine/StateMachineAssert.scala
package com.laamella.code_state_machine

import org.scalatest.Assertions._

import scala.collection.mutable

object StateMachineAssert {
  def assertActive[State, Event, Priority <: Ordered[Priority]](machine: StateMachine[State, Event, Priority], expectedStates: State*) {
    for (expectedState <- expectedStates) {
      if (!machine.active(expectedState)) {
        fail(s"Expected $expectedState to be active.")
      }
    }
    val expectedStatesSet = mutable.HashSet[State](expectedStates: _*)
    for (actualState <- machine.activeStates) {
      if (!expectedStatesSet.contains(actualState)) {
        fail(s"$actualState was active, but not expected.")
      }
    }
  }
}

File: java/src/test/java/com/laamella/code_state_machine/StateMachineAssert.java
package com.laamella.code_state_machine;

import java.util.Arrays;
import java.util.HashSet;

import static org.junit.jupiter.api.Assertions.fail;

public class StateMachineAssert {
    @SafeVarargs
    public static <T extends Enum<?>, E, P extends Comparable<P>> void assertActive( StateMachine<T, E, P> machine,  T... expectedStates) {
        for (var expectedState : expectedStates) {
            if (!machine.isActive(expectedState)) {
                fail("Expected " + expectedState + " to be active.");
            }
        }
        var expectedStatesSet = new HashSet<>(Arrays.asList(expectedStates));
        for (var actualState : machine.getActiveStates()) {
            if (!expectedStatesSet.contains(actualState)) {
                fail("" + actualState + " was active, but not expected.");
            }
        }
    }

}
